tools/docker/Dockerfile

```diff
@@ -0,0 +1,75 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2024, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This Dockerfile creates an image to run devlib CI tests.
+#
+# Running ``docker build -t devlib .`` command in ``tools/docker`` directory
+# creates the docker image.
+#
+# The image can be runned via ``docker run -it --privileged devlib`` command.
+#
+
+FROM ubuntu:22.04
+
+ENV DEBIAN_FRONTEND noninteractive
+
+ENV DEVLIB_REF master
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    aapt \
+    bc \
+    bison \
+    build-essential \
+    cmake \
+    cpio \
+    file \
+    flex \
+    git \
+    libelf-dev \
+    libncurses5-dev \
+    libssl-dev \
+    locales \
+    python3-pip \
+    qemu-system-arm \
+    qemu-system-x86 \
+    rsync \
+    sudo \
+    unzip \
+    wget \
+    vim \
+    xz-utils
+
+RUN apt-get -y autoremove && \
+    apt-get -y autoclean && \
+    apt-get clean && \
+    rm -rf /var/cache/apt
+
+RUN git clone -b ${DEVLIB_REF} -v https://github.com/ARM-software/devlib.git /devlib
+RUN cd /devlib && pip install .[full]
```

Comment:
`python setup.py install` is also obsolete. Unsure how to address your concern. Can you elaborate?

Reply:
> python setup.py install is also obsolete.

Calling setup.py directly is indeed obsolete, but `pip` calling it is very much still supported and is used by a large amount (the majority ?) of projects. `pyproject.toml` has gained a lot of traction recently, but it's quite new, and there are still things that require executing code (for example there is no way to automatically accumulate all the extra requires into an "all"/"full" extra requires, or complex builds with C extensions, things like that). There used to be a `setup.cfg` thing as well, but it's an older iteration of the same idea as `pyproject.toml`.

> Unsure how to address your concern. Can you elaborate?

```
pip install --upgrade pip setuptools wheels
pip install .[full]
```
If you don't do the first line, you might end up with:
* an antediluvian pip that does not know how to talk to current PyPI
* no setuptools package installed, leading to an ImportError in setup.py (assuming the setup.py uses setuptools, which it does for devlib and lots of projects)
* no wheels package, leading to issues if pip needs to build a wheel for whatever reason


Otherwise, if you migrate to using `pyproject.toml`, all you would need is ensuring pip is recent-enough, since the toml file describes what build backend is used (e.g. setuptools) so that pip can install it without having to do it manually beforehand:
```
pip install --upgrade pip
pip install .[full]
```

Reply:
Thank you! Updated.